ocaml/xapi/xapi_globs.ml

```diff
@@ -1574,6 +1582,18 @@ let host_virtual_hardware_platform_versions =
     has_vendor_device
   ]
 
+let recommendations t =
+  let list = match t with `Pool -> !pool_recommendations in
+  list
+  |> List.map (fun s ->
+         let k_v = String.split_on_char '=' s in
+         match k_v with
+         | [k; v] ->
+             (k, v)
+         | _ ->
+             failwith (Printf.sprintf "Parse recommendations failed, k_v: %s" s)
```

Comment:
I don't see this exception being handled anywhere in dbsync, this is very dangerous. Instead, instead use options along List.filter_map to ignore invalid values.

Reply:
If the error handling policy is to ignore parsing errors, then could the parsing be moved into "dbsync_master.ml"? It would be safe in this stage then.

Meanwhile, as @lindig suggested, a new module like `PoolRecommendation` and `PoolRecommendation.load dir` can be used. The `dir` can be a glob configuration in `xapi_globs.ml`, e.g. `/etc/xapi.recommendation.d/pool/`.